zed: small fixes

markgoddard opened the pull request:
- CI: Only sync new images after build
- swap: Note that Kayobe supports swap in Zed
- tempest: update max microversions for Cinder & Nova
- Update stackhpc.pulp to 0.5.2
- ceph: remove unnecesary Jinja braces
- aufn-ceph: Use stackhpc/zed branch in script
- requirements.txt: drop version specifier
- pulp: Stop syncing prometheus-jiralert image
- aufn-ceph: Drop cephadm_image override
- aufn-ceph: Sync inventory with kolla-ansible

markgoddard committed:
swap: Note that Kayobe supports swap in Zed

markgoddard committed:
tempest: update max microversions for Cinder & Nova

markgoddard committed:
Update stackhpc.pulp to 0.5.2

markgoddard committed:
ceph: remove unnecesary Jinja braces

markgoddard committed:
aufn-ceph: Use stackhpc/zed branch in script

markgoddard committed:
requirements.txt: drop version specifier

We should always be on Python 3.8+ in Zed.

markgoddard committed:
pulp: Stop syncing prometheus-jiralert image

We no longer support it.

markgoddard committed:
aufn-ceph: Drop cephadm_image override

The default in the base config should use this when Ceph images are not synced to Pulp.

markgoddard committed:
aufn-ceph: Sync inventory with kolla-ansible

markgoddard commented on .github/workflows/stackhpc-container-image-build.yml:
```diff
@@ -211,7 +211,8 @@ jobs:
           --ref main \
-          -f filter="$filter"
+          -f filter="$filter" \
+          -f sync-old-images=false
```
We could do the same for new images on yoga & earlier

markgoddard replied:
https://github.com/stackhpc/stackhpc-kayobe-config/pull/639

markgoddard commented on doc/source/configuration/walled-garden.rst:
```diff
@@ -62,6 +62,7 @@ proxy:
    ---
    # HTTP proxy URL (format: http(s)://[user:password@]proxy_name:port). By
    # default no proxy is used.
+   # FIXME: Need to apply the fix from 0e66f0b3dd08b470acb593977dce7dcb611cf5e3
```
Probably needs to go to yoga

markgoddard commented on etc/kayobe/kolla.yml:
```diff
@@ -331,14 +332,8 @@ kolla_build_customizations: "{{ kolla_build_customizations_common | combine(koll
 
 # Dict mapping Kolla Dockerfile ARG names to their values.
 kolla_build_args:
-  blackbox_exporter_version: "0.19.0"
```
These will need a rebuild & tag update

markgoddard replied:
I'll do this now

markgoddard replied:
Building: https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/6174749103

markgoddard replied:
Trying again... https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/6174929742

markgoddard replied:
Ubuntu builds failing :(

Alex-Welsh replied:
Test pulp is having issues. It failed on apt

markgoddard replied:
Or is that just the standard ubuntu repos getting out of sync issue? Will has a PR for yoga to bump the snapshot versions.

Alex-Welsh commented on doc/source/configuration/host-images.rst:
```diff
@@ -14,6 +14,8 @@ When configured, an image will be downloaded to the seed during the
 ``kayobe seed service deploy`` step, and subsequently deployed using bifrost
 with ``kayobe overcloud provision``.
 
+FIXME: Should we do this by default in Zed+?
```
Might be a bit redundant. Everyone has to migrate off cs8/rl8/focal on yoga anyway so should be using the images by then

Alex-Welsh replied:
But then it would make sense for new deployments to use them so perhaps it is worth it.

Alex-Welsh commented on doc/source/configuration/host-images.rst:
```diff
@@ -44,17 +46,16 @@ StackHPC Kayobe configuration provides configuration for some standard
 The overcloud DIB configuration is provided in
-``etc/kayobe/stackhpc-overcloud-dib.yml``. It is not used by default, and must
-be actively opted into. This can be done as follows:
+``etc/kayobe/stackhpc-overcloud-dib.yml``. It is used by default, but may be
```
Still needed in overcloud-dib.yml
```yaml
   overcloud_dib_host_images:
     - "{{ stackhpc_overcloud_dib_host_image }}"
```

Alex-Welsh commented on doc/source/contributor/environments/ci-aio.rst:
```diff
@@ -61,6 +61,7 @@ Install package dependencies when on Ubuntu:
 .. parsed-literal::
 
    sudo apt update
+   # FIXME: Stop using virtualenv?
```
I think about editing all of these out but other steps might have changed as well. I decided on leaving things as is and just adding a big warning until SMS came back and it could all actually be tested.

Alex-Welsh commented on etc/kayobe/ansible/requirements.yml:
```diff
@@ -3,7 +3,7 @@ collections:
     version: 1.14.0
   - name: stackhpc.pulp
-    version: 0.4.1
+    version: 0.5.2
```
Same for yoga? or is it another ansible/python version problem?

markgoddard replied:
I don't see anything incompatible. Might be good to keep yoga stable unless there's a specific fix we need though

Alex-Welsh commented on etc/kayobe/kolla.yml:
```diff
@@ -331,14 +332,8 @@ kolla_build_customizations: "{{ kolla_build_customizations_common | combine(koll
-  memcached_exporter_version: "0.9.0"
-  mysqld_exporter_version: "0.13.0"
-  node_exporter_version: "1.5.0"
+  node_exporter_version: "1.5.0" # kolla has 1.4.0
```
Pierre updated it in yoga 6 months ago to fix a crash issue

markgoddard replied:
oops, I didn't mean to commit the comment, that was a note to myself.

markgoddard replied:
Although actually it's kind of useful

Alex-Welsh commented on .automation:
What's the version change here?

markgoddard replied:
This was an accidental commit. Still a submodule noob.

Alex-Welsh commented on etc/kayobe/environments/aufn-ceph/kolla/inventory/overcloud-services.j2:
```diff
@@ -477,7 +488,7 @@ monitoring
 
 # NOTE(Alex-Welsh): This might need removing, since we have moved to opensearch
```
Time to remove this note?

markgoddard replied:
Done

Alex-Welsh commented on etc/kayobe/environments/ci-builder/globals.yml:
```diff
@@ -12,4 +12,4 @@ os_distribution: "{{ lookup('pipe', '. /etc/os-release && echo $ID') | trim }}"
 
 # Avoid a reboot.
-selinux_do_reboot: false
+selinux_state: disabled
```
Same for multinode?

markgoddard replied:
Done

markgoddard committed:
CI: Allow logging of Rally/Tempest

By default the 'Run tempest' task has no_log set to avoid revealing
sensitive data. This does not apply in CI, and makes it difficult to
debug failures.

Alex-Welsh commented on etc/kayobe/kolla/globals.yml:
```diff
@@ -10,12 +10,19 @@ enable_docker_repo: "{% raw %}{{ 'overcloud' not in group_names or ansible_facts
+# Each key is the tag variable prefix name, and the value is another dict,
+# where the key is the OS distro and the value is the tag to deploy.
+# NOTE: This is defined in etc/kayobe/kolla-image-tags.yml.
+kolla_image_tags: "{{ kolla_image_tags | to_nice_yaml(indent=2) }}"
```
How can this reference itself?

markgoddard replied:
Yeah it's a bit weird, but it's just a template, rather than a variable definition. See the variable in etc/kayobe/kolla-image-tags.yml.

Just trying it out really, thought it has two benefits:
* Avoid keeping the _tag variables in sync (I almost got the prometheus one wrong)
* Keep the globals.yml file more static. It often gets modified in client repos leading to collisions.

Alex-Welsh replied:
It might be useful in yoga, but I don't think we need to worry too much about changing it there and merging it up because it's release-specific and will conflict anyway

Alex-Welsh commented on etc/kayobe/kolla/globals.yml:
```diff
@@ -10,12 +10,20 @@ enable_docker_repo: "{% raw %}{{ 'overcloud' not in group_names or ansible_facts
-    ubuntu: zed-ubuntu-jammy-20230821T155947
-
-openstack_tag: "{% raw %}{{ kayobe_image_tags['openstack'][kolla_base_distro] }}{% endraw %}"
+# Dict of Kolla image tags to deploy for each service.
+# Each key is the tag variable prefix name, and the value is another dict,
+# where the key is the OS distro and the value is the tag to deploy.
+# NOTE: This is defined in etc/kayobe/kolla-image-tags.yml.
+kolla_image_tags:
+{{ kolla_image_tags | to_nice_yaml(indent=2) }}
+
+{% for tag in kolla_image_tags %}
+{% if tag == 'openstack' %}
+{{ tag }}_tag: "{% raw %}{{ kolla_image_tags['{% endraw %}{{ tag }}{% raw %}'][kolla_base_distro] }}{% endraw %}"
+{% else %}
+{{ tag }}_tag: "{% raw %}{{ kolla_image_tags['{% endraw %}{{ tag }}{% raw %}'][kolla_base_distro] | default(openstack_tag) }}{% endraw %}"
+{% endif %}
+{% endfor %}
```
Worth cherry picking to yoga?

markgoddard replied:
Possibly, just don't like to change things too much on old releases. Given the tag format is changing in zed anyway that seemed like a good point to change.

Alex-Welsh commented:
Might also be a good time to bump the pulp container a bit?

markgoddard commented:
> Might also be a good time to bump the pulp container a bit?

Could be, but needs testing.

Alex-Welsh commented:
`cephadm_image` in aufn-ceph/cephadm.yml needs updating

markgoddard commented:
> `cephadm_image` in aufn-ceph/cephadm.yml needs updating

I removed it since the default should now work.

Alex-Welsh commented:
Do you want more suggestions for things that need to change but are not included on this PR, or should we just try to get this through with the changes that are in at the moment?

Alex-Welsh commented:
> Do you want more suggestions for things that need to change but are not included on this PR, or should we just try to get this through with the changes that are in at the moment?

i.e. we need to change https://github.com/stackhpc/stackhpc-kayobe-config/blob/stackhpc/zed/.github/workflows/stackhpc-build-kayobe-image.yml#L70C12-L70C12 at some point to include `BASE_IMAGE: rockylinux:9`. There's also some references to `virtualenv` in the aufn deployment script that should be swapped for `python3 -m venv`

markgoddard commented:
> > Do you want more suggestions for things that need to change but are not included on this PR, or should we just try to get this through with the changes that are in at the moment?
> 
> i.e. we need to change https://github.com/stackhpc/stackhpc-kayobe-config/blob/stackhpc/zed/.github/workflows/stackhpc-build-kayobe-image.yml#L70C12-L70C12 at some point to include `BASE_IMAGE: rockylinux:9`. There's also some references to `virtualenv` in the aufn deployment script that should be swapped for `python3 -m venv`

Mostly looking for a +1/-1 on these changes. Unless there are issues with these or related things I'd prefer to get this one merged and make tickets in Jira for outstanding issues.
